lib/galaxy/datatypes/blast.py

```diff
@@ -102,7 +102,7 @@ def merge(split_files, output_file):
             # For one file only, use base class method (move/copy)
             return Text.merge(split_files, output_file)
         if not split_files:
-            raise ValueError("Given no BLAST XML files, %r, to merge into %s" % (split_files, output_file))
+            raise ValueError(f"Given no BLAST XML files, {split_files!r}, to merge into {output_file}")
```

Comment:
Puh, this looks not good.

Reply:
Sorry, why?

Reply:
```suggestion
            raise ValueError(f"Given no BLAST XML files, {repr(split_files)}, to merge into {output_file}")
```

https://peps.python.org/pep-0498/#s-r-and-a-are-redundant 
Just a suggestion - no strong opinion.

Reply:
No strong opinion either, but a quick test with `timeit` shows that using `repr()` is slower.

Reply:
Let's keep the `!r` syntax then. Thanks for running the test!